pipeline/pipeline.go

```diff
@@ -370,10 +371,18 @@ func returnModuleDataOutputs(clock *pbsubstreams.Clock, step bstream.StepType, c
 	return nil
 }
 
-func (p *Pipeline) renderWasmInputs(module *pbsubstreams.Module) (out []wasm.Argument, err error) {
+func (p *Pipeline) renderWasmInputs(module *pbsubstreams.Module, request *pbsubstreams.Request) (out []wasm.Argument, err error) {
 	storeAccessor := p.stores.StoreMap
 	for _, input := range module.Inputs {
 		switch in := input.Input.(type) {
+		case *pbsubstreams.Module_Input_Params_:
+			//TODO(colin): is this okay?
+			p := params.Decode(request.Params)
+			inputParams, ok := p[module.Name]
+			if !ok {
+				return nil, fmt.Errorf("params for module %q not found", module.Name)
+			}
+			out = append(out, wasm.NewParamsInput(input.GetParams().GetValue(), inputParams))
```

Comment:
Just that last line is needed, and with the `GetValue()`'s value.. no need for `inputParams`. See my comment below on the `NewParamsInput` function.